### ISSUE TITLE
Swift 5.9 + disabling redundantReturn error

### ISSUE THREAD

**Issue by JOyo246** [nicklockwood/SwiftFormat#1565]
When bumping our swift version to 5.9, and also disabling redundantReturn, we are seeing a bug with if/switch expressions.

Swift 5.8 Code
```swift
enum MyEnum {
    case a
    case b
}
let myEnum = MyEnum.a
let test: Int = {
    switch myEnum {
    case .a:
        return 0
    case .b:
        return 1
    }
}()
```

After bumping to 5.9, disabling redundantReturn and running swiftformat, we get...

```swift

enum MyEnum {
    case a
    case b
}
let myEnum = MyEnum.a
    
let test: Int = switch myEnum {
    case .a:
        return 0
    case .b:
        return 1
}
```

which is an error, Cannot 'return' in 'switch' when used as expression

**Comment by calda** [comment]
Here's a fix: https://github.com/nicklockwood/SwiftFormat/pull/1567

**Comment by nicklockwood** [comment]
@calda oh ha ha, I was just finishing up my own fix 🤦‍♂️

I think I prefer my approach, but I'll steal your test suite - thanks!

**Comment by calda** [comment]
Haha, thats funny, np! Will be interested to see your approach

**Comment by nicklockwood** [comment]
@JOyo246 fixed in 0.52.9
